src/peergos/server/storage/auth/BlockRequestAuthoriser.java

```diff
@@ -18,12 +18,38 @@ static boolean isValidAuth(BlockAuth auth, Cid block, Cid sourceNode, Bat bat, H
         S3Request req = new S3Request("GET", sourceNode.toBase58(), "api/v0/block/get?arg=" + block.toBase58(), S3Request.UNSIGNED,
                 Optional.of(auth.expirySeconds), false, true,
                 Collections.emptyMap(), Collections.emptyMap(), auth.batId.toBase58(), "eu-central-1", t);
-        Instant timestamp = Instant.parse(String.format("%s-%s-%sT%s:%s:%sZ", t.substring(0, 4), t.substring(4, 6), t.substring(6, 8), t.substring(9, 11), t.substring(11, 13), t.substring(13, 15)));
-        Instant expiry = timestamp.plusSeconds(auth.expirySeconds);
-        if (expiry.isBefore(Instant.now()))
+        LocalDateTime timestamp = LocalDateTime.of(
```

Comment:
have you tested this with leading zeros?

Reply:
It's parsing as an integer, so leading zeroes don't matter.

Reply:
done